### DISCUSSION TITLE
How to use Chatwoot with LINE integration and Google Dialogflow

### DISCUSSION THREAD

**Discussion by lasersonic** [chatwoot/chatwoot#5097] [Q&A] [answered]
Hi Guys,
Thank for awesome product. Now I have problem with the integration as following

I set up LINE with Google Dialogflow => worked, i do have chatbot on LINE channel.

I tried set up LINE with Chatwoot => worked, I do have live chat on LINE channel. 

But the question is I want chatbot to serve customer upfront by setting up NLU&flow by Google dialogflow. Then, if any further complex question, users can route to creat ticket in chatwoot.

I have done chatwoot integration with Dialogflow but it doesnt come to my expectation. Now I could only do chatbot only or live-chat only.

Dont know how to make it work...

Really thanks for help.

**Comment by sojan-official** [comment, marked as answer]
@lasersonic You can look at our agent bot APIs if you want more control over the experience. Instead of connecting your bot directly to LINE, your bot can interact with the messages in Chatwoot Inbox and create replies back into Chatwoot. Chatwoot will take care sending and receiving messages to LINE.

ref: https://www.chatwoot.com/docs/product/others/agent-bots/

**Reply by lasersonic** [reply]
Thank you let me explore on that!
